plugins/sqs/sqs.cpp

```diff
@@ -0,0 +1,539 @@
+          cq_out.GetError().GetMessage() << std::endl;
+     }
+
+delete cq_req;
+return ref;
+}
+
+/**
+
+
+
+ **/
+SQSHPCCPlugin::Response SQSHPCCPlugin::SQSHPCC::receiveMessage() 
+{
+
+   SQSHPCCPlugin::Response ref = {};
+
+try{
+  
+  Aws::SQS::Model::ReceiveMessageRequest receiveMessageRequest;
+  receiveMessageRequest.SetQueueUrl(this->queueUrl);
+  receiveMessageRequest.SetMaxNumberOfMessages(1);
+  receiveMessageRequest.AddMessageAttributeNames("All");
+
+  Aws::SQS::Model::ReceiveMessageOutcome receiveMessageOutcome = this->sqsClient->ReceiveMessage(receiveMessageRequest);
+ if(!receiveMessageOutcome.IsSuccess() || receiveMessageOutcome.GetResult().GetMessages().size() == 0) {
+       std::cout << "Error on receive: " << receiveMessageOutcome.GetError().GetMessage() << std::endl;
+       ref.code=2;
+       return ref;
+ }
+ 
+ Aws::SQS::Model::Message msg = receiveMessageOutcome.GetResult().GetMessages()[0];
+ cout << msg.GetBody() << endl;
+} catch(const char* message) {
+
+  cout << "Error occurs " << message << endl; 
+}
+
+ return ref;
+}
+
+
+/**
+ *
+ *
+ *
+ *
+ **/
+void SQSHPCCPlugin::SQSHPCC::setQueueUrlFromQueueName() 
+{
+
+   Aws::SQS::Model::GetQueueUrlRequest* gqu_req = new Aws::SQS::Model::GetQueueUrlRequest();
+   gqu_req->SetQueueName(this->queueName.c_str());
+
+try {
+   Aws::SQS::Model::GetQueueUrlOutcome gqu_out = this->sqsClient->GetQueueUrl(*gqu_req);
+//const char* bl = gqu_out.IsSuccess() ? "true":"false";
+//Aws::String strBl(bl);
+//this->queueUrl=strBl;
+  if(gqu_out.IsSuccess()) {
+        std::cout << "Queue " << this->queueName.c_str() << " has url " << std::endl;
+    this->queueUrl=gqu_out.GetResult().GetQueueUrl();
+   } else {
+        std::cout << "Error getting url for queue " << this->queueName.c_str() << ": " << std::endl;
+      throw runtime_error(gqu_out.GetError().GetMessage().c_str()) ;
+    // this->queueUrl = gqu_out.GetError().GetMessage();
+   }
+
+ } catch(const char* message) {
+     cout << "Error occurs " << message << endl;
+}
+
+ delete(gqu_req);
+}
+
+
+/**
+
+
+ **/
+void SQSHPCCPlugin::SQSHPCC::setSQSConfiguration(const string& protocol, const string& region)
+ {
+
+  Aws::InitAPI(this->options);
+  Aws::Client::ClientConfiguration* config= new Aws::Client::ClientConfiguration();
+ if(!protocol.empty())
+  {
+    config->scheme = Aws::Http::Scheme::HTTPS;
+  }
+
+  if(region.empty()) throw string("Region mustn't be empty");
+  if(isRegionExist(region))
+    {
+      config->region = getRegion(region);
+    }
+
+  this->sqsClient= new Aws::SQS::SQSClient(*config);
+
+ //this->queueUrl=getRegion(region);
+   cout << "Queue URL is "  << this->queueUrl << endl;
+// this->setQueueUrlFromQueueName();
+ 
+ delete(config);
+}
+
+
+
+/**
+
+
+ **/
+bool SQSHPCCPlugin::SQSHPCC::isRegionExist(const string& region)
+{
+  
+  if(region.empty())
+    {
+      return false;
+    }
+
+  char *reg = new char[region.length() +1];
```

Comment:
This code feels like it should be using strieq (or an equivalent method in class string, if there is one) rather than copying to a new string and upper-casing just to do a case-insensitive compare.